### ISSUE TITLE
Saving space on the default Spellbook Tables

### ISSUE THREAD

**Issue by kgar** [kgar/foundry-vtt-tidy-5e-sheets#293]
related to #286 

With Tidy set to its default settings, we are out of space on the Spellbook lists.

![image](https://github.com/kgar/foundry-vtt-tidy-5e-sheets/assets/4632129/0fb0c59d-f3c9-44e6-8346-841aa3c4d188)

## Recovering Some Space

If possible, we need to recover some space in this list without taking away critical information that is needed during high-stakes scenarios like combat, chases, etc.

### Refine Spellbook First Column Overflow Control

The name column is currently the victim when the item tables get too compressed, causing the name to progressively disappear. However, in the Spell tables, the spell pips and tracker seem to prevent from being hidden, causing the misalignment issues.

Update the spellbook name column to hide its contents progressively as it shrinks in size, but when the user hovers over the column, allow the contents to show fully, with absolute positioning, over the rest of the grid, with a solid background.

### Kiligir's Idea: Spell School Icons

Community discord reference: https://discord.com/channels/1167985253072257115/1170004050578767973/1191594432982687754

Regarding the spell schools on the spell sheet, I think it'd look nice if they were instead replaced by [RPG Awesome icons](https://nagoshiashumari.github.io/Rpg-Awesome/). Namely:

- Abjuration: ra-shield - A shield, because protection.
- Conjuration: ra-tentacle - A tentacle, because monster.
- Divination: ra-crystal-ball - A crystal ball, because scry.
- Enchantment: ra-aware  - A dude with lines around his head, because mind control.
- Evocation: ra-burning-meteor - A flaming ball of rock, because ow.
- Illusion: ra-bleeding-eye  - A stylish eye, because see things wrong.
- Necromancy: ra-skull - A skull, because the Ethical Necromancers Association would take my ENA card if we didn't.
- Transmutation: ra-cycle - A cycle, because it's turning things into other things. This is the only one I'm iffy on. I wanted to do a potion bottle because transmutation was about alchemy IRL.

Some questions to answer if we take this route:

- Do we provide a way for custom spell schools to specify an icon at the world level?
- What should we show when a custom spell school is present but not accounted for?

### Classic Controls Overhaul

#96

**Comment by crash1115** [comment]
> transmutation icon

When I saw this, I expected sheep first, potion bottle second. :joy: Either of those would be immediately recognizable to me, but I get the arrows too.

> Do we provide a way for custom spell schools to specify an icon at the world level?

If we can specify custom schools, then I think it tracks that there should be a way to specify icons. Whether that's a custom image, or just a picker that lets you choose from some selection of pre-canned icons, doesn't particularly matter to me. The benefit of the latter is that it'll ensure at least a small degree of uniformity, and i think that's at least kinda important in a module like thiss one.

> What should we show when a custom spell school is present but not accounted for?

I'd suggest sparkles as the "sane default": ✨

**Comment by kgar** [comment]
Discussion about spell pips in relation to dnd5e 3.x default sheets:
https://discord.com/channels/1167985253072257115/1170021717524107274/1209578587301085204

**Comment by kgar** [comment]
Suggestion:
use the new default sheets approach of hiding nonessential columns in priority order, based on the sheet width.

**Comment by kgar** [comment]
Closing this issue. Future efforts will come from the item table overhaul and a new (eventual replacement) spellbook tab
